frontend/src/main/scala/bloop/data/SettingsMapper.scala

```diff
@@ -0,0 +1,106 @@
+package bloop.data
+import bloop.engine.BuildLoader
+import bloop.logging.Logger
+import bloop.logging.DebugFilter
+import java.nio.charset.StandardCharsets
+import bloop.config.ConfigEncoderDecoders
+import io.circe.Printer
+import io.circe.Json
+import io.circe.parser
+import bloop.io.AbsolutePath
+import java.nio.file.Files
+import bloop.config.Config.Settings
+import io.circe.JsonObject
+import _root_.bloop.config.Config
+import bloop.DependencyResolution
+import scala.util.Try
+
+object SettingsMapper {
+
+  /**
+    * @param settingsPath Path to the json settings file.
+    * @param logger The logger that collects messages about loading the settings.
+    * @return Settings if the file exists.
+    */
+  def fromFile(settingsPath: AbsolutePath, logger: Logger): Option[Settings] = {
+    if (settingsPath.isFile) {
+      val bytes = Files.readAllBytes(settingsPath.underlying)
+      logger.debug(s"Loading workspace settings from ${BuildLoader.settingsFileName}")(
+        DebugFilter.All
+      )
+      val contents = new String(bytes, StandardCharsets.UTF_8)
+      parser.parse(contents) match {
+        case Left(failure) => throw failure
+        case Right(json) => Some(fromJson(json))
+      }
+    } else {
+      None
```

Comment:
I will do it in another place, since we here if the file is not present it might mean we are actually running it not from Metals and we don't want to add SemanticDB plugin in that case